langkit/support/langkit_support-token_data_handlers.adb

```diff
@@ -681,16 +688,117 @@ package body Langkit_Support.Token_Data_Handlers is
    begin
       --  Return slice from...
       return
-        TDH.Source_Buffer (
-         --  The first character in the requested line
-         TDH.Lines_Starts.Get (Line_Number)
+        TDH.Source_Buffer
+          (
+           --  The first character in the requested line
+           TDH.Lines_Starts.Get (Line_Number)
+
+           ..
 
-         ..
+           --  The character before the LF that precedes the first character of
+           --  the next line.
+             TDH.Lines_Starts.Get (Line_Number + 1) - 2
+          );
 
-         --  The character before the LF that precedes the first character of
-         --  the next line.
-           TDH.Lines_Starts.Get (Line_Number + 1) - 2
-        );
    end Get_Line;
 
+   --------------
+   -- Get_Sloc --
+   --------------
+
+   function Get_Sloc
+     (TDH : Token_Data_Handler; Offset : Natural) return Source_Location
+   is
+      function Compare
+        (Sought      : Positive;
+         Dummy_Index : Positive;
+         Line_Start  : Positive) return Relative_Position
+      is
+        (if Sought > Line_Start then After
+         elsif Sought = Line_Start then Inside
+         else Before);
+
+      function Get_Line_Index is new Floor (Positive, Index_Vectors);
```

Comment:
I find it hard to infer from its implementation what this function is supposed to compute. Please add a docstring, such as:
> Given the index of a codepoint in TDH.Source_Buffer and given TDH.Line_Starts, return the index in TDH.Source_Buffer of the first character of the line that contains this codepoint.

Reply:
OK. Although needs to be said I copied from the other instantiations in this package, that don't have a comment and are not much clearer ... Seems OK to me because the generic function is here in the same package and documented. (Also this one has at least a descriptive name).

Reply:
I’m pretty sure I’m the author of the other instantiation, yet no problem if think it’s not clear and ask me to document it. :-) Someone else’s eyes helping spotting missing comments and all…

IIRC, the reason I asked here is that even though I understood what the generic was supposed to do (which tells me more or less *how* Get_Line_Index is implemented, not *why*), I had I had trouble getting what was the purpose of `Get_Line_Index`, even when looking at the call site.